src/types/line.js

```diff
@@ -1,10 +1,10 @@
 import {Element, defaults} from 'chart.js';
-import {isArray, fontString, toRadians} from 'chart.js/helpers';
+import {isArray, toFontString, toRadians} from 'chart.js/helpers';
 import {scaleValue, roundedRect, inTriangle} from '../helpers';
 
 const pointInLine = (p1, p2, t) => ({x: p1.x + t * (p2.x - p1.x), y: p1.y + t * (p2.y - p1.y)});
-const interpolateX = (y, p1, p2) => pointInLine(p1, p2, Math.abs((y - p1.y) / (p2.y - p1.y))).x;
-const interpolateY = (x, p1, p2) => pointInLine(p1, p2, Math.abs((x - p1.x) / (p2.x - p1.x))).y;
+const interpolateX = (y, p1, p2) => pointInLine(p1, p2, y / (p2.y + p1.y)).x;
```

Comment:
oh, now I realize this does not work.

Say the Y of p1 is 50, and p2 is 100. Now if we want to interpolate the x at Y=50, we need to have the 3rd parameter be 0. `50 / (100 + 50)` is not 0. `(50-50)/(100-50)` however is. Also at Y=100, the 3rd parameter needs to be 1. 100/150 is not 1.

Where did the original go wrong?

Reply:
Well, I'm sure that I have found the previous interpolation didn't work... But now, I don't remember that use case...
Let me revert  to previous interpolation. If  I'll find the issue again, I'll open an issue.

Reply:
Ignore my previous post... The reason of the change was in the issue #269

Reply:
In the master (therefore in new version and only there) if, configuring a line annotation for the label, you set the `position` , to `top, bottom, left or right`, the label is not shown.
It works only with `center`.

It seems an issue on interpolation calculation in the `calculateLabelPosition` of line annotation.

https://github.com/chartjs/chartjs-plugin-annotation/blob/da07bf955e00523bc8912e5017a36af7cd8c68f9/src/types/line.js#L8-L9

That's the reason and with new "interpolation" I'm quite sure that works.

Reply:
It works with original interpolation. Let me revert to it.